src/macports1.0/sysctl.c

```diff
@@ -92,7 +91,36 @@ int SysctlCmd(ClientData clientData UNUSED, Tcl_Interp *interp, int objc UNUSED,
     Tcl_SetObjResult(interp, tcl_result);
     return TCL_OK;
 #else
-    Tcl_SetObjResult(interp, Tcl_NewStringObj("sysctl not available", -1));
-    return TCL_ERROR;
+    char *name;
+    name = Tcl_GetString(objv[1]);
+    if (objc != 2) {
+        Tcl_WrongNumArgs(interp, 1, objv, "name");
+        return TCL_ERROR;
+    }
```

Comment:
Shouldn't you check `objc` before accessing `objv[1]`?

```suggestion
    char *name;
    if (objc != 2) {
        Tcl_WrongNumArgs(interp, 1, objv, "name");
        return TCL_ERROR;
    }

    name = Tcl_GetString(objv[1]);
```